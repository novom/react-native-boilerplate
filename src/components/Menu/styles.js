import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F5FCFF',
  },

  headerContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    height: 150,
  },

  headerText: {
    fontSize: 36,
    color: '#CCC',
  },
});
