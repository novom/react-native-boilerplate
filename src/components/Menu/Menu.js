import React from 'react';
import { Text, View } from 'react-native';
import { DrawerView } from 'react-navigation';

import styles from './styles';

/**
 * Container displayed into the drawer
 * The menu items is generated by DrawerView
 */
const Menu = props => (
  <View style={styles.container}>
    <View style={styles.headerContainer}>
      <Text style={styles.headerText}>HEADER</Text>
    </View>
    <DrawerView.Items {...props} />
  </View>
);

export default Menu;
