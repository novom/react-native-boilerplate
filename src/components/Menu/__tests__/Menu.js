/* eslint-disable no-undef */

import 'react-native';

import React from 'react';
import renderer from 'react-test-renderer';

import Menu from '../Menu';

const defaultProps = {
  getLabel: () => {},
  navigation: {
    state: {
      routes: [
        { key: 'Home', routeName: 'Home' },
        { key: 'About', routeName: 'About' },
      ],
    },
  },
  renderIcon: () => {},
};

it('renders menu correctly', () => {
  const tree = renderer.create(
    <Menu {...defaultProps} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
