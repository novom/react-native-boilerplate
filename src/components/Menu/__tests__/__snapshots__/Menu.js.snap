// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders menu correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 150,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#CCC",
          "fontSize": 36,
        }
      }
    >
      HEADER
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 20,
          "paddingVertical": 4,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
